Lib/pyclbr.py

```diff
@@ -58,31 +59,36 @@ def __init__(self, module, name, file, lineno, parent):
         self.lineno = lineno
         self.parent = parent
         self.children = {}
+        if parent is not None:
+            parent._addchild(name, self)
 
     def _addchild(self, name, obj):
         self.children[name] = obj
 
 
 class Function(_Object):
     "Information about a Python function, including methods."
-    def __init__(self, module, name, file, lineno, parent=None):
-        _Object.__init__(self, module, name, file, lineno, parent)
-
+    def __init__(self, module, name, file, lineno, parent=None, is_async=False):
+        super().__init__(module, name, file, lineno, parent)
+        self.is_async = is_async
+        if isinstance(parent, Class):
+            parent._addmethod(name, lineno)
 
 class Class(_Object):
     "Information about a Python class."
-    def __init__(self, module, name, super, file, lineno, parent=None):
-        _Object.__init__(self, module, name, file, lineno, parent)
-        self.super = [] if super is None else super
+    def __init__(self, module, name, super_, file, lineno, parent=None):
+        super().__init__(module, name, file, lineno, parent)
+        self.super = super_ or []
         self.methods = {}
 
     def _addmethod(self, name, lineno):
         self.methods[name] = lineno
 
-
-def _nest_function(ob, func_name, lineno):
+# This 2 functions are used in these tests
+# Lib/test/test_pyclbr, Lib/idlelib/idle_test/test_browser.py
+def _nest_function(ob, func_name, lineno, is_async=False):
     "Return a Function after nesting within ob."
-    newfunc = Function(ob.module, func_name, ob.file, lineno, ob)
+    newfunc = Function(ob.module, func_name, ob.file, lineno, ob, is_async)
     ob._addchild(func_name, newfunc)
```

Comment:
If _Object and Function do what _addchild and _addmethod do, calling them is redundant.  As is turns out, `dic[key] = ob` is idempotent, so the 2nd call does not actually hurt.